test/gmp-thread.jl

```diff
@@ -0,0 +1,9 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+x = BigInt()
+
+# This will cause a SIGSEGV prior to https://github.com/JuliaLang/julia/pull/27020
+ret = @threadcall((:__gmpz_init, :libgmp), Void, (Ptr{BigInt}, ), Ref(x))
+
+# Not really needed, as long as it doesn't segfault, we're ok
+@test typeof(ret) == Void
```

Comment:
On 0.7 `Void` is called `Nothing` but for C interop there is also the typealias `Cvoid`. You would also need a `using Test` in this file but I think it is better to move it to some existing file, probably https://github.com/JuliaLang/julia/blob/master/test/threads.jl.

You can also write this as:

```jl
@testset "#27020 SIGSEV bignum init" begin
    x = BigInt()
    ret = @threadcall((:__gmpz_init, :libgmp), Cvoid, (Ptr{BigInt}, ), Ref(x))
    @test typeof(ret) == Cvoid
end
```

Reply:
thanks, have incorporated all your changes.